src/requests.c

```diff
@@ -282,8 +282,14 @@ JsonNode *torrent_add_from_file(gchar *target, gint flags)
 
     json_object_set_boolean_member(args, PARAM_PAUSED, (flags & TORRENT_ADD_FLAG_PAUSED));
 
-    if ((flags & TORRENT_ADD_FLAG_DELETE))
-        g_unlink(target);
+    if ((flags & TORRENT_ADD_FLAG_DELETE)) {
+        g_autoptr(GFile) file = NULL;
+        g_autoptr(GError) error = NULL;
+
+        file = g_file_new_for_path(target);
+        if (!g_file_trash(file, NULL, &error))
+            g_warning("Failed to trash %s: %s", target, error->message);
```

Comment:
I'm not familiar with how trash cans work, the docs just say something about "if the file system supports it", but I definitely don't use a file system that comes with that feature. I also don't use a DE like GNOME/KDE. So my question is will this lead to spurious log spam?

Reply:
This is the implementation for systems without gvfs installed:
https://gitlab.gnome.org/GNOME/glib/blob/main/gio/glocalfile.c#L1965
so shouldn't cause problems on any reasonably setup systems. You're probably in a better place to test it than I am though.